crates/sui-core/src/authority.rs

```diff
@@ -635,16 +635,13 @@ impl AuthorityState {
 
         self.metrics.total_cert_attempts.inc();
 
-        if certificate.contains_shared_object()
-            && !epoch_store.is_tx_cert_consensus_message_processed(certificate)?
-        {
-            return Err(SuiError::CertificateNotSequencedError {
-                digest: *certificate.digest(),
-            });
+        if !certificate.contains_shared_object() {
+            // Shared object transactions will be sequenced by Narwhal eventually.
+            // So rely on the submission to consensus to execute the certificate.
```

Comment:
nit: how about "Shared object transactions need to be sequenced by Narwhal before enqueueing for execution. They are done in `handle_consensus_transaction()`."